client/src/pages/ManageBookings/ManageBookings.tsx

```diff
@@ -16,39 +17,33 @@ import { useHistory } from 'react-router-dom';
 import useStyles from './useStyles';
 
 const ManageBookings = (): JSX.Element => {
-  const { loggedInUser } = useAuth();
   const { initSocket } = useSocket();
 
-  const history = useHistory();
   const classes = useStyles();
 
   useEffect(() => {
     initSocket();
   }, [initSocket]);
 
-  const [date, changeDate] = useState<MaterialUiPickersDate>(new Date());
+  const [date, setDate] = useState<MaterialUiPickersDate>(new Date());
   const [isSelectedDate, setIsSelectedDate] = useState<boolean>(false);
 
   const [requests, setRequests] = useState<Array<BookingRequest>>(mockBookingRequest);
-  const nextBooking = requests.reduce((request, closest) =>
-    new Date(request.start.getTime() - new Date().getTime()) < closest.start &&
-    request.start &&
-    request.start > new Date()
-      ? (closest = request)
-      : (closest = closest),
-  );
-  const [selectedBooking, setSelectedBooking] = useState<BookingRequest | undefined>(nextBooking);
 
-  if (loggedInUser === undefined) return <CircularProgress />;
-  if (!loggedInUser) {
-    history.push('/login');
-    // loading for a split seconds until history.push works
-    return <CircularProgress />;
-  }
+  const nextBooking = requests
+    .filter((request) => request.start.getTime() - new Date().getTime() > 0)
+    .reduce(
+      (request, closest) =>
+        closest.start.getTime() - new Date().getTime() > request.start.getTime() - new Date().getTime()
+          ? (closest = request)
+          : (closest = closest),
```

Comment:
Reduce assigns the return value of this callback to closest automatically. We can remove these assignments and return just the rhs:
```suggestion
          ? request
          : closest,
```